play-json/shared/src/main/scala/play/api/libs/json/Reads.scala

```diff
@@ -31,7 +31,8 @@ import ScalaCollectionCompat._
 @implicitNotFound(
   "No Json deserializer found for type ${A}. Try to implement an implicit Reads or Format for this type."
 )
-trait Reads[A] { self =>
+@SerialVersionUID(2338626292552178385L)
+trait Reads[A] extends Serializable { self =>
```

Comment:
This is a sign there is some reference to a Reads that is leaking, and is not a good sign... I checked the branch locally and the problem is in JsNull. If you add a `@transient` to the `@transient implicit val reads` and remove this code the test is green in scala 2.12...

Reply:
done